src/compass/utils/h5_helpers.py

```diff
@@ -561,43 +561,44 @@ def corrections_to_h5group(parent_group, burst, cfg):
-
-    # create linspace for axises shared by both LUTs
-    x_end = bistatic_delay_lut.x_start + bistatic_delay_lut.width * bistatic_delay_lut.x_spacing
-    slant_range = np.linspace(bistatic_delay_lut.x_start, x_end,
-                              bistatic_delay_lut.width, dtype=np.float64)
-    y_end = bistatic_delay_lut.y_start + bistatic_delay_lut.length * bistatic_delay_lut.y_spacing
-    azimuth = np.linspace(bistatic_delay_lut.y_start, y_end,
-                          bistatic_delay_lut.length, dtype=np.float64)
-
-    # correction LUTs axis and doppler correction LUTs
-    desc = ' correction as a function of slant range and azimuth time'
-    correction_items = [
-        Meta('slant_range', slant_range, 'slant range of LUT data',
-             {'units': 'meters'}),
-        Meta('slant_range_spacing', bistatic_delay_lut.x_spacing,
-             'spacing of slant range of LUT data', {'units': 'meters'}),
-        Meta('zero_doppler_time', azimuth, 'azimuth time of LUT data',
-             {'units': 'seconds'}),
-        Meta('zero_doppler_time_spacing', bistatic_delay_lut.y_spacing,
-             'spacing of azimuth time of LUT data', {'units': 'seconds'}),
-        Meta('bistatic_delay', bistatic_delay_lut.data,
-             f'bistatic delay (azimuth) {desc}', {'units': 'seconds'}),
-        Meta('geometry_steering_doppler', geometrical_steering_doppler.data,
-             f'geometry steering doppler (range) {desc}',
-             {'units': 'meters'}),
-        Meta('azimuth_fm_rate_mismatch', az_fm_mismatch.data,
-             f'azimuth FM rate mismatch mitigation (azimuth) {desc}',
-             {'units': 'seconds'}),
-    ]
-    for meta_item in correction_items:
-        add_dataset_and_attrs(correction_group, meta_item)
+    # If enabled, save the correction LUTs
+    if cfg.lut_params.enabled:
+        geometrical_steering_doppler, bistatic_delay_lut, az_fm_mismatch = \
+            compute_geocoding_correction_luts(burst,
+                                              dem_path=cfg.dem,
+                                              rg_step=cfg.lut_params.range_spacing,
+                                              az_step=cfg.lut_params.azimuth_spacing)
+
+        # create slant range and azimuth vectors shared by the LUTs
+        x_end = bistatic_delay_lut.x_start + bistatic_delay_lut.width * bistatic_delay_lut.x_spacing
+        slant_range = np.linspace(bistatic_delay_lut.x_start, x_end,
+                                  bistatic_delay_lut.width, dtype=np.float64)
+        y_end = bistatic_delay_lut.y_start + bistatic_delay_lut.length * bistatic_delay_lut.y_spacing
+        azimuth = np.linspace(bistatic_delay_lut.y_start, y_end,
+                              bistatic_delay_lut.length, dtype=np.float64)
+
+        # correction LUTs axis and doppler correction LUTs
+        desc = ' correction as a function of slant range and azimuth time'
```

Comment:
Can we just delete this leading space? looks like it's adding 2 spaces in the final string
```
{'description': 'azimuth FM rate mismatch mitigation (azimuth)  correction as a function of slant range and azimuth time'}
```

```suggestion
        desc = 'correction as a function of slant range and azimuth time'
```

Reply:
Oh and the only extra question I had while looking at it was where are the units stored? Might be a question for @LiangJYu , but I wasn't seeing the units in any of my poking around with xarray

```python

In [1]: import datatree

In [2]: ds = datatree.open_datatree("stack-pr78/t078_165494_iw2/20220713/t078_165494_iw2_20220713_VV.h5", engine='h5netcdf', phony_dims='sort')

In [3]: da = ds['science/SENTINEL1/CSLC/corrections']['azimuth_fm_rate_mismatch']


In [22]: da
Out[22]:
<xarray.DataArray 'azimuth_fm_rate_mismatch' (phony_dim_0: 111, phony_dim_1: 485)>
[53835 values with dtype=timedelta64[ns]]
Dimensions without coordinates: phony_dim_0, phony_dim_1
Attributes:
    description:  azimuth FM rate mismatch mitigation (azimuth)  correction a...


In [29]: da.attrs
Out[29]: {'description': 'azimuth FM rate mismatch mitigation (azimuth)  correction as a function of slant range and azimuth time'}

Reply:
@scottstanie I've assigned #83 to myself to keep tabs on this.